### PR TITLE
Fixed scalings in likelihood computation

### PR DESCRIPTION
**Pull request by JereKoskela**
I've fixed the scaling inconsistency that Ali pointed out. The likelihood function arguments are now per-generation mutation and recombination probabilities, as well as Ne, and everything is calculated in generations. 

I think I've kept everything consistent with your overall plumbing and all tests pass in both C and python, but let me know if you spot anything that looks out of place @jeromekelleher.

### CODE REVIEW

File: msprime/likelihood.py [hunk 1]
```diff
@@ -60,9 +60,9 @@ def unnormalised_log_mutation_likelihood(arg, theta):
     return ret
 
 
-def log_arg_likelihood(arg, recombination_rate, Ne=0.25):
+def log_arg_likelihood(arg, recombination_rate, Ne=0.5):
```
Comment (jeromekelleher) [inline msprime/likelihood.py]: We'll need to change the default to 1 before release --- is there much work in changing the tests?

Comment (JereKoskela) [reply msprime/likelihood.py]: Nah, it shouldn't take me too long. I'll probably not have a chance to get to it this week, but will get it done shortly thereafter.

Comment (jeromekelleher) [reply msprime/likelihood.py]: Great, let's get that in with this PR so. Thanks.

Comment (JereKoskela) [reply msprime/likelihood.py]: Having looked at it, it turns out that the tests don't use the default value of Ne. As of the latest commit, they all feed in an explicit argument of 0.5, so we can change the default without affecting tests in any way. Would that be ok, or do you think we need a more comprehensive overhaul?

Comment (jeromekelleher) [reply msprime/likelihood.py]: No, great, let's just change the default and write a couple of quick tests to check the default is what we think it is.

**Comment by jeromekelleher** [comment]
Hey @JereKoskela, do you think we can finish this up please? We can open an issue to track any remaining problems, but it'd be nice to get the code merged soon. (We want to make some style changes to the codebase, so want to minimise the number of open PRs)

**Comment by JereKoskela** [comment]
It looks like I changed the default `Ne` in the update of Jan 29th. Apologies for not nudging you at the time @jeromekelleher. I don't really understand the codecov complaints in the checks, but apart from that I think this is complete.

**Comment by jeromekelleher** [comment]
Ah, OK, thanks @JereKoskela. Can you do a quick rebase so please to bring this up to date and I'll merge then.

**Comment by JereKoskela** [comment]
I think that has worked. Merge away @jeromekelleher.

**Comment by jeromekelleher** [comment]
I've added some tests to and updated the code to cover some edge cases @JereKoskela - can you take a quick look to see if you agree please?

File: lib/tests/tests.c [hunk 2]
```diff
@@ -2084,6 +2084,122 @@ test_gene_conversion_simulation(void)
...
+    /* Zero edges should give a likelihood of zero */
+    ret = msp_log_likelihood_arg(&ts, 1, 1, &lik);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_EQUAL_FATAL(lik, 0);
+    ret = msp_log_likelihood_arg(&ts, 0, 1, &lik);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    CU_ASSERT_EQUAL_FATAL(lik, 0);
+
+    /* TODO why do we need this special case? Is it an error or does this
+     * return value have some significance? */
```
Comment (JereKoskela) [inline lib/tests/tests.c]: This is returning a zero likelihood (-infinite log likelihood) when there are in the ARG but the putative mutation rate is zero. I was future-proofing for things like grid search or MCMC for parameter values.

Comment (jeromekelleher) [reply lib/tests/tests.c]: Ah, I see, thanks.. I'll update the code (we should document this when we're writing up the docs too).

File: tests/test_likelihood.py [hunk 3]
```diff
@@ -30,8 +30,9 @@
 # Python implementations of the likelihood calculations
 
 
-# TODO give this the same signature as the production version
-def log_arg_likelihood(arg, r, Ne):
+def log_arg_likelihood(arg, recombination_rate, Ne=1):
+    if recombination_rate <= 0:
+        raise ValueError("Recombination rate must be > 0")
```
Comment (JereKoskela) [inline tests/test_likelihood.py]: Do we want/need to raise an error for a zero recombination rate? There's no mathematical reason for it: the likelihood is positive with zero recombination rate if there are no breaks in the ARG, and zero if there are breaks.

File: lib/tests/tests.c (hunk 2 above)
Comment (jeromekelleher) [reply lib/tests/tests.c]: Should we also return -inf with a zero recombination rate? Seems like a reasonable thing to do?

Comment (JereKoskela) [reply lib/tests/tests.c]: Yes, the documentation is on my to-do list, and finally getting rather high on it too.

Comment (JereKoskela) [reply lib/tests/tests.c]: > Should we also return -inf with a zero recombination rate? Seems like a reasonable thing to do?

-inf when recombination rate is zero and there are break points in the ARG. If there are no break points, then the existing implementation will return a finite number when the recombination rate is zero.

File: tests/test_likelihood.py (hunk 3 above)
Comment (jeromekelleher) [reply tests/test_likelihood.py]: I've no real opinion here, just trying to cover the corner cases. How about you follow up here with a commit which deals with the boundaries at zero?

Comment (jeromekelleher) [reply tests/test_likelihood.py]: You should be able to run ``git pull`` on your local branch and it'll pull the changes from your ``origin`` remote.

File: lib/tests/tests.c (hunk 2 above)
Comment (jeromekelleher) [reply lib/tests/tests.c]: > I'll update the code (we should document this when we're writing up the docs too).

If you're updating the code, it would make sense for you to delete this comment while you're in there.

File: tests/test_likelihood.py (hunk 3 above)
Comment (JereKoskela) [reply tests/test_likelihood.py]: Thanks, I'll have a look later on this evening, and delete the mutation comment while I'm at it.

**Comment by JereKoskela** [comment]
Hi @jeromekelleher. I successfully updated my local files via `git pull`. The C code compiles nicely and all the C tests pass, but `nosetests` for the python code throws a bunch of TypeErrors and ultimately segmentation faults before I have changed anything. Any idea what might be going on? I see that my likelihood branch is 10 commits behind again; would another rebase help?

**Comment by jeromekelleher** [comment]
> The C code compiles nicely and all the C tests pass, but nosetests for the python code throws a bunch of TypeErrors and ultimately segmentation faults before I have changed anything. Any idea what might be going on?

You probably just need to compile your low-level python module by running ``make`` in the top level directory.

> I see that my likelihood branch is 10 commits behind again; would another rebase help?

Yes, sorry about that - things are moving quickly at the moment! Can you rebase please? You'll hit a conflict on the circleci yaml file which will hopefully be easy enough to resolve. It's probably a good idea to push your more recent commit first though, so we have a backup (I can always do the rebasing then if needs be).

**Comment by JereKoskela** [comment]
> > The C code compiles nicely and all the C tests pass, but nosetests for the python code throws a bunch of TypeErrors and ultimately segmentation faults before I have changed anything. Any idea what might be going on?
> 
> You probably just need to compile your low-level python module by running `make` in the top level directory.

Thanks, that worked! I haven't made any more changes yet, so I'll try rebasing now and then fixing the zero rate cases. There's a good chance I'll be back for help with resolving the conflict.

**Comment by JereKoskela** [comment]
All sorted @jeromekelleher. I'm happy for this to be merged if you are.

**Comment by jeromekelleher** [comment]
Merged; thanks @JereKoskela!
